### ISSUE TITLE
[nomad] Constraint did not meet topology requirement

### ISSUE THREAD

**Issue by GiyoMoon** [hetznercloud/csi-driver#771]
### TL;DR

I followed the [nomad guide](https://github.com/hetznercloud/csi-driver/tree/main/docs/nomad) and when trying to add a job that makes use of the volumes I get the error "Constraint did not meet topology requirement filtered 1 node".

### Expected behavior

The job should be able to use the volume. Maybe it's a bug with the csi driver or the nomad guide is missing something.

### Observed behavior

Any jobs that want to make use of a volume can't be created due to constraints not meeting topology requirements.

### Minimal working example

Follow the [nomad guide](https://github.com/hetznercloud/csi-driver/tree/main/docs/nomad) 1:1 with the **latest** csi driver version (v2.10.0) and you should encounter this error.

### Log output

```plain text
+ Job: "volume-name"
+ Task Group: "group-name" (1 create)
  + Task: "task-name" (forces create)

Scheduler dry-run:
- WARNING: Failed to place all allocations.
  Task Group "group-name" (failed to place 1 allocation):
    * Constraint "did not meet topology requirement": 1 nodes excluded by filter

Job Modify Index: 0
To submit the job with version verification run:

nomad job run -check-index 0 test.hcl

When running the job with the check-index flag, the job will only be run if the
job modify index given matches the server-side version. If the index has
changed, another user has modified the job and the plan's results are
potentially invalid.
```

### Additional information

```
nomad - 1.9.1
hcloud-csi-driver - v2.10.0
```

**Comment by GiyoMoon** [comment]
When printing the volume information, I get:
```
ID                   = db-vol
Name                 = db-vol
Namespace            = default
External ID          = REDACTED
Plugin ID            = csi.hetzner.cloud
Provider             = csi.hetzner.cloud
Version              = 2.10.0
Capacity             = 10 GiB
Schedulable          = true
Controllers Healthy  = 1
Controllers Expected = 1
Nodes Healthy        = 1
Nodes Expected       = 1
Access Mode          = <none>
Attachment Mode      = <none>
Mount Options        = fs_type: ext4 flags: [REDACTED]
Namespace            = default

Topologies
Topology  Segments
01        csi.hetzner.cloud/location=nbg1

Allocations
No allocations placed
```

Maybe my node needs be assigned to the nbg1 location somehow to meet the constraint? If yes I don't know how to add this, any help would be appreciated 😇 Thanks!

**Comment by resmo** [comment]
I am the author of the doc and can confirm this issue after updating to csi 2.10.0. Fallback to 2.9.0 resolved the issue for me.

**Comment by lopcode** [comment]
Bumped in to the same issue today, confirmed downgrading to 2.9.0 worked 🤞

**Comment by resmo** [comment]
I briefly run over the commits and the changes in this PR look suspicious to me: #743

Pinging @apricote to let him know.

**Comment by GiyoMoon** [comment]
@resmo Hey, thanks for looking into this! I can confirm that 2.9.0 works for me too.

**Comment by lukasmetzner** [comment]
Hi,

we are in the process of building a Nomad development setup to run csi-driver end-to-end tests on Nomad. During this setup, we encountered this bug. We are currently investigating the issue.

Best Regards
Lukas

**Comment by lukasmetzner** [comment]
Hi,

we have just released [v2.10.1](https://github.com/hetznercloud/csi-driver/pull/779) to revert the breaking changes.

We apologize for any inconvenience this may have caused.

Best regards,
Lukas

**Comment by GiyoMoon** [comment]
Hey Lukas, thanks! Can confirm that `2.10.1` works :)
